arangod/Cluster/Maintenance.cpp

```diff
@@ -315,7 +314,7 @@ static void handlePlanShard(
   bool shouldBeLeading = serverId == leaderId;
   bool replication2Leader =
       replicationVersion == replication::Version::TWO &&
-      isReplication2Leader(dbname, shname, localLogs, shardsToLogs);
+      isReplication2Leader(shname, localLogs.value(), shardsToLogs.value());
```

Comment:
This is influenced just by internal logic, indeed. The `replicationVersion == replication::Version::TWO` condition should be all we need to avoid accessing null optionals. For replication2 databases, there should always be valid `localLogs` and `shardsToLogs`, even if they are empty maps.
In case we'll make errors on our side, the `.value()` call will throw `bad_optional_access`, which is caught by the maintenance and logged on level `ERR`, so we'll find out.